content/academy/puppeteer_playwright/page.md

```diff
@@ -0,0 +1,311 @@
+import { chromium } from 'playwright';
+
+const browser = await chromium.launch({ headless: false });
+
+const page = await browser.newPage();
+
+await browser.close();
+</marked-tab>
+<marked-tab header="Puppeteeer" lang="javascript">
+import puppeteer from 'puppeteer';
+
+const browser = await puppeteer.launch({ headless: false });
+
+const page = await browser.newPage();
+
+await browser.close();
+</marked-tab>
+```
+
+Then, we can visit a website with the `page.goto()` method. Let's go to [Google](https://google.com) for now. We'll also create a custom function called `sleep`, which will force the program to wait for a number of seconds before moving quitting (otherwise, everything will just flash before our eyes and we won't really tell what's going on):
+
+```marked-tabs
+<marked-tab header="Playwright" lang="javascript">
+import { chromium } from 'playwright';
+
+const sleep = (secs) => new Promise((r) => setTimeout(r, secs * 1000));
+
+const browser = await chromium.launch({ headless: false });
+
+const page = await browser.newPage();
+
+await page.goto('https://google.com')
+
+// wait for 10 seconds before shutting down
+await sleep(10);
+
+await browser.close();
+</marked-tab>
+<marked-tab header="Puppeteeer" lang="javascript">
+import puppeteer from 'puppeteer';
+
+const sleep = (secs) => new Promise((r) => setTimeout(r, secs * 1000));
+
+const browser = await puppeteer.launch({ headless: false });
+
+const page = await browser.newPage();
+
+await page.goto('https://google.com');
+
+// wait for 10 seconds before shutting down
+await sleep(10);
+
+await browser.close();
+</marked-tab>
+```
+
+> If you haven't already, go ahead and run this code.
+
+## [](#interacting-with-a-page) Interacting with a page
+
+The **Page** object has whole boat-load of functions which can be used to interact with the loaded page. We're not going to go over all of them, but we _will_ use a few very important ones to add some functionality to our current project.
+
+Let's say that we want automate searching for **hello world** on Google, then click on the first result and log the title of the page to the console, then take a screenshot and write it it to the filesystem. In order to understand how we're going to automate this, let's break down how we would do it manually:
+
+1. Click on the button which accepts Google's cookies policy
+2. Type **hello world** into the search bar
+3. Press **Enter**
+4. Wait for the results page to load
+5. Click on the first result
+6. Read the title of the result's page
+7. Screenshot the page
+
+Though it seems complex, the wonderful **Page** API makes all of these actions extremely easy to perform.
+
+### Clicking & pressing keys
+
+Let's first focus on the first 3 steps listed above. By using `page.click()` and the CSS selector of the element to click, we can click an element:
+
+```JavaScript
+// Click the "I understand" button
+await page.click('button + button:nth-child(2)');
```

Comment:
Playwright can click based on content. Maybe even Puppeteer can now. https://playwright.dev/docs/selectors This is really good for action buttons like "Log in" or "I understand" as the text is less likely to change than the CSS.